project-reports/2023/2023-Q4-Hyperledger-FireFly.md

```diff
@@ -0,0 +1,117 @@
+- Correctly parse blockchain subscription names when delivering events
+
+## [FireFly v1.2.1](https://github.com/hyperledger/firefly/releases/tag/v1.2.1)
+
+FireFly v1.2.1 is a patch release that contains many fixes, including:
+
+- Fixed postgres optimizations, fail gracefully when inserting many data rows
+- Replace “UpsertBatch” with “InsertOrGetBatch”
+- Enable workflows for release branch
+- Backport fixes
+- Fixed multiple named tuple result from contract
+
+# Overall Activity in the Past Quarter
+
+The major highlights have been detailed at the beginning of the report. In summary, there has been a lot of activity in the project. New community members continue to discover the project and make contributions.
+
+# Current Plans
+
+## FireFly v1.3.0
+
+The next feature release of FireFly is currently in active development and is expected to be available in early Q1 2024. It will include many new exciting features:
+
+- Significantly enhanced onchain / offchain data coordination including the ability to attach a FireFly message to a custom contract invocation
+- Additional APIs for deleting FireFly Resources such as Token Pools, Contract Interfaces etc.
+- Configurable mTLS security across FireFly microservices for internal service-to-service authentication and encryption
+- PostgreSQL support for EVMConnect persistence
+- ERC-1155 token connector improvements for supporting a wider variety of ERC-1155 contracts
+- Performance improvements in transaction throughput and latency
+
+An additional, significant architectural roadmap item is active-active HA for FF FireFly Core. This will offer scaling improvements, increased resiliency, and more flexible deployment and migration options. This work is currently in the design phase and the related work is being tracked in GitHub.
+
+# Maintainer Diversity
+
+FireFly now has 12 maintainers representing three different companies:
+
+- [Kaleido](https://kaleido.io/)
+- [Fidelity](https://www.fidelity.com/)
+- [OneOf](https://www.oneof.com/)
+
+It is worth noting that each company has at least one maintainer who is a subject matter expert on one or more specific blockchain connectors, and the most foundational parts of the FireFly technology stack.
+
+A list of all maintainers for Hyperledger FireFly may be found [here](https://wiki.hyperledger.org/display/FIR/Maintainers).
+
+# Contributor Diversity
+
+In the past quarter we’ve seen an increase in contributor strength, commit growth, and new contributors from parties outside of Kaleido.
+
+[Commit Activities](https://insights.lfx.linuxfoundation.org/projects/hyperledger%2Ffirefly/dashboard;subTab=technical;v=source-control%2Fcommits%2Foverview)
+
+![alt_text](images/HLFF_2023_Q4_1.png "image_tooltip")
+
+![alt_text](images/HLFF_2023_Q4_2.png "image_tooltip")
+
+![alt_text](images/HLFF_2023_Q4_3.png "image_tooltip")
+
+![alt_text](images/HLFF_2023_Q4_4.png "image_tooltip")
+
+![alt_text](images/HLFF_2023_Q4_5.png "image_tooltip")
```

Comment:
Do you have insights to measure maintainers activity across organizations similar to contributor activities?

Reply:
The list of maintainers is here: https://wiki.hyperledger.org/display/FIR/Maintainers
Currently 10 from Kaleido, 1 from Fidelity, and 1 from Instinct Tools.